### PR TITLE
Updating docs with references to kind property

### PR DESCRIPTION
**Pull request by AndrewRTsao**
Also adding more details in reference sections along with a page around structuring your project (via file explorer)

### CODE REVIEW

**Comment by AndrewRTsao** [comment]
@ericpgreen2 For your review but we should hold off on pushing this docs update until we actually release this feature

**Comment by AndrewRTsao** [comment]
Just removed `-- @kind: model` from our examples so our docs should be finalized now

**Review by ericpgreen2 — APPROVED** [review]
Looks great!

File: docs/docs/reference/project-files/index.md [hunk 1]
```diff
@@ -7,11 +7,17 @@ hide_table_of_contents: true
...
+:::info Working with resources outside their native folders
+
+It is possible to define resources (such as [sources](sources.md), [models](models.md), [dashboards](dashboards.md), [custom APIs](apis.md), or [themes](themes.md)) within <u>any</u> nested folder within your Rill project directory. However, for any YAML configuration file, it is then imperative that the `kind` property is then appropriately defined within the underlying resource configuration or Rill will not able to resolve the resource type correctly!
+
+:::
 
 Projects can simply be rehydrated from Rill project files into an explorable data application as long as there is sufficient access and credentials to the source data - figuring out the dependencies, pulling down data, & validating your model queries and metrics configurations. The result is a set of functioning exploratory dashboards.
 
-You can see an example project by visiting our [example github repository](https://github.com/rilldata/rill-developer-example.git).
+You can see a few different example projects by visiting our [example github repository](https://github.com/rilldata/rill-developer-example.git).
```
[the remark this answers is not in the document]
Comment (AndrewRTsao) [reply docs/docs/reference/project-files/index.md:20]: Good catch

**Comment by AndrewRTsao** [comment]
Docs should be finalized and good to go

**Pull request merged**